src/core/public/kbn_bootstrap.ts

```diff
@@ -27,60 +27,35 @@
 
 import { i18n } from '@kbn/i18n';
 import { CoreSystem } from './core_system';
+import { ApmSystem } from './apm_system';
 
 /** @internal */
-export function __kbnBootstrap__() {
+export async function __kbnBootstrap__() {
   const injectedMetadata = JSON.parse(
     document.querySelector('kbn-injected-metadata')!.getAttribute('data')!
   );
 
-  /**
-   * `apmConfig` would be populated with relavant APM RUM agent
-   * configuration if server is started with `ELASTIC_APM_ACTIVE=true`
-   */
-  const apmConfig = injectedMetadata.vars.apmConfig;
-  const APM_ENABLED = process.env.IS_KIBANA_DISTRIBUTABLE !== 'true' && apmConfig != null;
+  const apmSystem = new ApmSystem(injectedMetadata.vars.apmConfig);
+  await apmSystem.setup();
 
-  if (APM_ENABLED) {
-    // eslint-disable-next-line @typescript-eslint/no-var-requires
-    const { init, apm } = require('@elastic/apm-rum');
-    if (apmConfig.globalLabels) {
-      apm.addLabels(apmConfig.globalLabels);
-    }
-    init(apmConfig);
+  let i18nError: Error | undefined;
+  try {
+    await i18n.load(injectedMetadata.i18n.translationsUrl);
```

Comment:
we can run i18n and apm-agent loading in parallel, but I don't think an error during `'@elastic/apm-rum'` loading should lead to a global error as i18n does. btw should we mute an error for `'@elastic/apm-rum'` failed loading?

Reply:
Parallel makes sense 👍 Could we also call `coreSystem.setup` in parallel? I suspect there may be an implicit dependency that the i18n module is initialized before setup, so maybe not.

Agreed, any errors from APM configuration should not block starting up Kibana. I think logging a warning would be nice for debugging purposes.

Reply:
> Parallel makes sense

@joshdover added. PTAL. 

>  Could we also call coreSystem.setup in parallel

We shouldn't as we may want to instrument `setup` with apm agent.